### PR TITLE
add AST to WordsSay

### LINKED ISSUE #16: Say doesn't take an AST.
Make it so Say can take an AST.

### PR DESCRIPTION
**Pull request by zwdnyc**
resolve #16 
1. add AST to WordsSay
2. move WordsProgramExceptions to words.exceptions

### CODE REVIEW

**Comment by alexthemark** [comment]
Do you know what's causing your blank lines to appear weird in the diff? It makes it hard to read the PR. Otherwise, looks good to me.

**Comment by zwdnyc** [comment]
My editor auto deletes trailing whitespaces while saving. Sorry for any inconvenience.

Simply adding ?w=1 to the end of the URL will cause GitHub to ignore whitespace for all the diffs on the page.

File: src/words/environment/WordsSay.java [hunk 1]
```diff
@@ -1,22 +1,49 @@
...
-	public WordsSay(String message) {
-		this.message = message;
+	private AST messageExpression;
+	private String messageValue;
+
+	public WordsSay(AST messageExpression) {
+		this.messageExpression = messageExpression;
+	}
+
+	public WordsSay(String messageValue) {
```
Comment (akallem) [inline src/words/environment/WordsSay.java]: I don't think we should allow this form of construction anymore.  This was the old, incorrect way of doing it.  The grammar only allows an expression, so I think we should do the same here.  That means some existing tests will need to be modified, but that's OK and should be easy to change.

File: src/words/environment/WordsSay.java [hunk 2]
```diff
@@ -1,22 +1,49 @@
...
 	@Override
 	public boolean isExecutable() {
 		return true;
 	}
 
 	@Override
-	protected void doExecute(WordsObject object, WordsEnvironment environment) {
-		object.setMessage(message);
+	protected void doExecute(WordsObject object, WordsEnvironment environment) throws WordsProgramException {
+		if (messageExpression != null) {
```
Comment (akallem) [inline src/words/environment/WordsSay.java]: Per above, I think this should be changed that the only way to construct it is with an expression, in which case this check isn't necessary (but we could have an assertion for it).

**Comment by alexthemark** [comment]
Cool trick on the whitespace, thanks!

**Comment by zwdnyc** [comment]
WordsSay only takes AST now.

**Comment by akallem** [comment]
OK, this looks good to me now.

**Comment by mzbenami** [comment]
looks good to me, too.

**Pull request merged**